### DISCUSSION TITLE
WAT! No shortnames for pilot-points in tpl files?

### DISCUSSION THREAD

**Discussion by laat0003** [pypest/pyemu#540] [Q&A] [answered]
Hi there,

I'm using longnames=False in PstFrom and for some wacky reason when I build Pst the template files for my pilot-points are not being updated with the shortnames. I have zone, constant and grid pars in this parameterization setup and they all get the short names replaced in their respective template files. The obs names in the ins files are also being replaced as expected and the mapping text files are written. Is there any reason why it might ignore the pilot-point files?

Cheers,

**Comment by jtwhite79** [comment, marked as answer]
Sounds buggy - we will take a look...

**Reply by laat0003** [reply]
All good - thanks. I wasn't sure if it was a known issue, work in progress or if I had to be using a separate template file directory. Do you want me to raise it in issues?

**Reply by jtwhite79** [reply]
not a known issue!  please open an issue - that will put more pressure on us to address it...

**Reply by briochh** [reply]
Hey @laat0003 , def not a known issue. My last set up I was using pilot points with short names and I think it was ok. I switched to using pypestutils however which plumbs in slightly differently so it is possible that something has broken in the pyemu pp renaming. 

That said, that rename function shouldn't care how pars are set up. It just rips through tpl and ins files mapping to new names. Could be a pathing issue. Def something we should look into.

**Reply by laat0003** [reply]
I think I figured out what the underlying problem is. I don't have multipliers for my pilot-points in the ./mult folder. Instead, I seem to be ending up with a .fac file, a .var.dat file, and a tpl file, that writes the .var.dat all of which are in my main working folder. I think this is related to me messing around with pypestutils while pyemu was out of town but unsure how she found out at this point. Once a player...

**Reply by laat0003** [reply]
Did some more digging. Installed new environment and tested different pp setups. Looks like the error occurs when passing in a Pandas dataframe of predefined points. It adds the pars and creates the required files but fails trying to do the build when it replaces long names for short names. My previous env didn't actually fail - no idea why. I'll raise the issue. Thanks for putting up with me :)
